File: 198. House Robber.java
//198. House Robber


class Solution {
    public int rob(int[] nums) {
        int max=0,sum=0;
        for(int i:nums){
            int temp=Math.max(max,sum+i);
            sum=max;
            max=temp;
        }
        return max;
    }
}

File: 198. House Robber.js
//198. House Robber


/**
 * @param {number[]} nums
 * @return {number}
 */
var rob = function(nums) {
    let max=0,sum=0;
    for(let num of nums){
        let temp=Math.max(max,sum+num);
        sum=max;
        max=temp;
    }
    return max;
};
